esmvaltool/diag_scripts/shared/iris_helpers.py

```diff
@@ -106,45 +106,75 @@ def convert_to_iris(dict_):
     dict
         Converted dictionary.
 
+    Raises
+    ------
+    KeyError
+        :obj:`dict` contains keys``'short_name'`` **and** ``'var_name'``.
+
     """
     dict_ = dict(dict_)
     if 'short_name' in dict_:
         if 'var_name' in dict_:
-            logger.warning(
-                "Dictionary already contains 'var_name', replacing old value "
-                "'%s' by '%s'", dict_['var_name'], dict_['short_name'])
+            raise KeyError(
+                f"Cannot replace 'short_name' by 'var_name', dictionary "
+                f"already contains 'var_name' (short_name = "
+                f"'{dict_['short_name']}', var_name = '{dict_['var_name']}')")
         dict_['var_name'] = dict_.pop('short_name')
     return dict_
 
 
-def iris_project_constraint(projects, cfg, negate=False):
-    """Create `iris.Constraint` to select specific projects from data.
+def get_mean_cube(datasets):
```

Comment:
This looks very close to the `multi_model_statistics` preprocessor function, do we need an extra function here?

Reply:
Yes, this is similar to `multi_model_statistics` but way easier. Also, `multi_model_statistics` takes `products` as input, which is inconvenient in this case. I'd like to keep that.

Reply:
Multi model statistics also supports cubes since a few months. Actually, I hope that multi model statistics will soon look like this function, once we have lazy vertical interpolation I think it can be implemented like that. However, it's fine if you want to keep it like this, we can always replace this later. When I was first reviewing this, I thought it was part of the public API, but it isn't, so it should be fine to add and remove things.